common/src/stack/command/stack.py

```diff
@@ -24,117 +24,131 @@
+			submodpath = '/'.join(fullmodpath[2:])
+			try:
+				help.run({'subdir': submodpath}, [])
+			except CommandError as e:
+				sys.stderr.write('%s\n' % e)
+				return -1
+			print(help.getText())
+			return -1
 
-	if not hasattr(module, 'Command'):
-		import stack.commands.list.help
-		help = stack.commands.list.help.Command(db)
-		fullmodpath = s.split('.')
-		submodpath = '/'.join(fullmodpath[2:])
 		try:
-			help.run({'subdir': submodpath}, [])
+			command = getattr(module, 'Command')(self.db, debug=debug)
+			rc = command.runWrapper(name, args[i:])
 		except CommandError as e:
 			sys.stderr.write('%s\n' % e)
+			syslog.syslog(syslog.LOG_ERR, '%s' % e)
+			return -1
+		except Exception as e:
+			# Sanitize Exceptions, and log them.
+			exc, msg, tb = sys.exc_info()
+			for line in traceback.format_tb(tb):
+				syslog.syslog(syslog.LOG_DEBUG, '%s' % line)
+				sys.stderr.write(line)
+			error = '%s: %s -- %s' % (module.__name__, exc.__name__, msg)
+			sys.stderr.write('%s\n' % error)
+			syslog.syslog(syslog.LOG_ERR, error)
 			return -1
-		print(help.getText())
-		return -1
-
-	try:
-		command = getattr(module, 'Command')(db, debug=debug)
-		rc = command.runWrapper(name, args[i:])
-	except CommandError as e:
-		sys.stderr.write('%s\n' % e)
-		syslog.syslog(syslog.LOG_ERR, '%s' % e)
-		return -1
-	except Exception as e:
-		# Sanitize Exceptions, and log them.
-		exc, msg, tb = sys.exc_info()
-		for line in traceback.format_tb(tb):
-			syslog.syslog(syslog.LOG_DEBUG, '%s' % line)
-			sys.stderr.write(line)
-		error = '%s: %s -- %s' % (module.__name__, exc.__name__, msg)
-		sys.stderr.write('%s\n' % error)
-		syslog.syslog(syslog.LOG_ERR, error)
-		return -1
 
-	text = command.getText()
+		text = command.getText()
 
-	# set the SIGPIPE to the system default (instead of python default)
-	# before trying to print; prevents a stacktrace when exiting a pipe'd stack command
-	signal.signal(signal.SIGPIPE, signal.SIG_DFL)
+		# set the SIGPIPE to the system default (instead of python default)
+		# before trying to print; prevents a stacktrace when exiting a pipe'd stack command
+		signal.signal(signal.SIGPIPE, signal.SIG_DFL)
 
-	if text and len(text) > 0:
-		print(text, end='')
-		if text[len(text) - 1] != '\n':
-			print()
-	syslog.closelog()
-	if rc is True:
-		return 0
-	return -1
+		if text and len(text) > 0:
+			print(text, end='')
+			if text[len(text) - 1] != '\n':
+				print()
+		syslog.closelog()
+		if rc is True:
+			return 0
+		return -1
 
 
 
 # attach a prettier interrupt handler to SIGINT (ctrl-c)
 signal.signal(signal.SIGINT, sigint_handler)
 
 # Open syslog
-
 syslog.openlog('SCL', syslog.LOG_PID, syslog.LOG_LOCAL0)
 
 
-db  = get_mysql_connection()
+runner = StackRunner()
 
 try:
 	opts, args = getopt.getopt(sys.argv[1:], '', ['debug', 'help', 'version'])
 except getopt.GetoptError as msg:
 	sys.stderr.write("error - %s\n" % msg)
-	if db is not None:
-		db.close()
+	if runner.db is not None and runner.db.open:
+		runner.db.close()
```

Comment:
Why aren't you letting stack runner clean up itself here?

Reply:
Fixed this